### PR TITLE
Add configuration option for disk filtering using predicates

### LINKED ISSUE #42: Is it possible to ignore a disk?
We are setting up a restor kiosk that lays down an autodmg image. I was wondering if it's possible to ignore a disk in Restor? We have a 10 TB disk that we are using to take a snapshot of the TDM drive before we wipe and restor it, but restor tries to restor the 10 TB disk :P

### PR DESCRIPTION
**Pull request by russellhancox**
Remove DiskFilter class, replace with a default predicate in ConfigController with the ability to add a second predicate in the configuration profile.
Add command-line option --debug-disk-filters to make debugging this config easier.
Add a few more properties to Disk class and a proper description method

Fixes #42

### CODE REVIEW

File: Common/Disk.m [hunk 1]
```diff
@@ -53,6 +55,13 @@ - (NSUInteger)hash {
   return result;
 }
 
+- (NSString *)description {
+  return [NSString stringWithFormat:@"bsd:%@, volName:%@, volKind:%@, whole:%d, "
+                                     "internal: %d, vendor: %@, model: %@",
```
Comment (tburgin) [inline Common/Disk.m]: be consistent with space / no-space between : and %@

File: Common/Disk.m [hunk 2]
```diff
@@ -53,6 +55,13 @@ - (NSUInteger)hash {
   return result;
 }
 
+- (NSString *)description {
+  return [NSString stringWithFormat:@"bsd:%@, volName:%@, volKind:%@, whole:%d, "
```
Comment (tburgin) [inline Common/Disk.m:59]: what is the dangling `"`?

**Review by tburgin — DISMISSED** [review]

File: Common/Disk.m (hunk 2 above)
Comment (russellhancox) [reply Common/Disk.m:59]: It's not dangling, it's a continuation

File: Common/Disk.m (hunk 1 above)
Comment (russellhancox) [reply Common/Disk.m]: Done

File: Common/Disk.m (hunk 2 above)
Comment (tburgin) [reply Common/Disk.m:59]: oops, I see it now

**Review by tburgin — APPROVED** [review]

**Pull request merged**